src/Parser/Parts/PartList.cs

```diff
@@ -10,6 +10,23 @@ public PartList()
         {
         }
 
+ 
+        
+        /// <summary>
+        /// Number of <see cref="TextPart"/> with type <see cref="HoleType.Numeric"/>
+        /// </summary>
+        public int NumericPartCount { get; set; }
```

Comment:
Not sure about that code quote. From the doc comment it seems this is an API to manually construct a template from tokens, and it validates that your tokens are either all positional or none.

Makes sense in that context.

I did settle for a `IsPositional` flag on the list after parsing. So no consistency issue. The flag is trivially true for a template with no hole.